docs/building-spark.md

```diff
@@ -79,15 +81,23 @@ Example:
 
     ./build/mvn -Pyarn -Dhadoop.version=2.8.5 -DskipTests clean package
 
+    ./build/mvn -Pyarn -Phadoop-3.2 -Dhadoop.version=3.1.2 -DskipTests clean package
+
 ## Building With Hive and JDBC Support
 
 To enable Hive integration for Spark SQL along with its JDBC server and CLI,
 add the `-Phive` and `Phive-thriftserver` profiles to your existing build options.
-By default Spark will build with Hive 1.2.1 bindings.
+By default, Spark will build hadoop-2.7 with Hive 1.2.1 and hadoop-3.2 with Hive 2.3.5.
 
     # With Hive 1.2.1 support
     ./build/mvn -Pyarn -Phive -Phive-thriftserver -DskipTests clean package
 
+    # With Hive 2.3.5 support
+    ./build/mvn -Pyarn -Phive -Phive-thriftserver -Phadoop-3.2 -DskipTests clean package
+
+    # Hadoop 2.7 with Hive 2.3.5 support
+    ./build/mvn -Pyarn -Phive -Phive-thriftserver -Phadoop-3.2 -Dhadoop.version=2.7.4 -Dcurator.version=2.7.1 -Dzookeeper.version=3.4.6 -DskipTests clean package
```

Comment:
Users can do this, but this looks too risky to me. I'd like to recommend not to advertise this combination `Using Hadoop 2.7.4 version under hadoop-3.2 profile`.

cc @gatorsmile , @cloud-fan , @srowen

Reply:
Yes, we don't need to spell out every possible combination especially unconventional ones. I would not add this clause.

Reply:
OK. removed it.